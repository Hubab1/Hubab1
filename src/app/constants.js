export const community = {
    name: '555 Waverly',
};

export const ROUTES = {
    // Independent routes
    COMMUNITY: '/',
    WELCOME: '/welcome',
    LOGIN: '/login',
    SIGNUP: '/signup',
    ADDRESS: '/address',
    MANUAL_INCOME_ENTRY: '/manual-income-entry',
    FEES_AND_DEPOSITS: '/fees-deposits',
    HOLDING_DEPOSIT_AGREEMENT: '/holding-deposit-agreement', // For HelloSign HD Agreement
    HOLDING_DEPOSIT_TERMS_AGREEMENT: '/holding-deposit-terms-agreement', // For payment terms agreement
    SCREENING: '/screening',
    APP_COMPLETE: '/application-complete',
    LEASE_TERMS: '/lease-terms',
    OUTSTANDING_BALANCE: '/outstanding-balance',
    PAYMENT_TERMS: '/payment-terms',
    ACCOUNT: '/account',
    RESEND_INVITE: '/resend-invite',
    APP_APPROVED: '/application-approved',
    LEASE_SIGNED: '/lease-signed',
    LEASE_EXECUTED: '/lease-executed',
    APP_DENIED: '/application-denied',
    TERMS: '/terms',
    APP_CANCELLED: '/application-cancelled',
    LEASE_VOIDED: '/lease-voided',
    PRIVACY_POLICY: '/privacy-policy',
    FUNNEL_TERMS: '/funnel-terms',
    FAQ: '/faq',
    UNIT_UNAVAILABLE: '/unit-unavailable',
    UNAUTHENTICATED_UNIT_UNAVAILABLE: '/unauth-unit-unavailable',
    GUARANTOR_REQUESTED: '/guarantor_request',
    PAYMENT_DETAILS: '/payment-details',

    // Password related routes
    PASSWORD: '/password',
    FORGOT_PASSWORD: '/password/forgot',
    VERIFY_PASSWORD_CODE: '/password/verify',
    RESET_PASSWORD: '/password/reset',

    // Rental Profile related routes
    RENTAL_PROFILE: '/rental-profile',
    PROFILE_OPTIONS: '/rental-profile/options',
    GUARANTOR: '/rental-profile/invite-guarantor',
    PARKING: '/rental-profile/parking',
    PETS: '/rental-profile/pets',
    STORAGE: '/rental-profile/storage',
    WINE_COOLER: '/rental-profile/wine-cooler',
    CO_APPLICANTS: '/rental-profile/invite-roommates',
    EDIT_DEPENDANT: '/rental-profile/dependent/:id/edit',
    REMOVE_PERSON: '/rental-profile/:type/:id/remove',

    // Income and Employment related routes
    INCOME_AND_EMPLOYMENT: '/income-employment',
    BANKING: '/income-employment',
    INCOME_VERIFICATION_SUMMARY: '/income-employment/summary',
    MANUAL_INCOME_ENTRY_ADD_INCOME: '/income-employment/manual/add-income',
    MANUAL_ASSET_ENTRY_ADD_ASSET: '/income-employment/manual/add-asset',
    EDIT_MANUAL_FINANCIAL_SOURCE: '/income-employment/manual/financial-sources/:id/edit',
    REMOVE_FINANCIAL_SOURCE: '/income-employment/manual/financial-sources/:id/remove',
    EMPLOYER_DETAILS: '/income-employment/employer-details',

    // Applications
    APPLICATIONS: '/applications',
};

export const ROUTE_LABELS = {
    [ROUTES.ADDRESS]: 'Current Address',
    [ROUTES.LEASE_TERMS]: 'Lease Terms',
    [ROUTES.PROFILE_OPTIONS]: 'Rental Profile',
    [ROUTES.INCOME_AND_EMPLOYMENT]: 'Income & Employment',
    [ROUTES.FEES_AND_DEPOSITS]: 'Fees & Deposits',
    [ROUTES.HOLDING_DEPOSIT_AGREEMENT]: 'Holding Deposit Agreement',
    [ROUTES.SCREENING]: 'Screening',
    [ROUTES.APP_COMPLETE]: 'Application Complete',
    [ROUTES.PAYMENT_DETAILS]: 'Payment Details',
    [ROUTES.HOLDING_DEPOSIT_TERMS_AGREEMENT]: 'Holding Deposit Agreement',
};

export const ROUTES_TOP_LEVEL = [
    ROUTES.ADDRESS,
    ROUTES.LEASE_TERMS,
    ROUTES.PROFILE_OPTIONS,
    ROUTES.INCOME_AND_EMPLOYMENT,
    ROUTES.FEES_AND_DEPOSITS,
    ROUTES.SCREENING,
    ROUTES.APP_COMPLETE,
    ROUTES.HOLDING_DEPOSIT_AGREEMENT,
    ROUTES.OUTSTANDING_BALANCE,
];

export const DOES_NOT_EXIST = 'DoesNotExist';
export const LEASE_TERMS_IDENTIFIER = 'lease_terms';
export const RENTER_PROFILE_IDENTIFIER = 'renter_profile';
export const UNIT_NOT_HELD_WAITING_IDENTIFIER = 'unit_not_held_waiting';

export const REPORT_POLL_INTERVAL = 10000;

export const STRIPE_PUBLISHABLE_KEY_DEMO = 'pk_test_vWO9q0lWvTCjPSq2FKMQeTpv00zsjtEm5j';
export const STRIPE_PUBLISHABLE_KEY_LIVE = 'pk_live_96Wm4LI5GkUtEFjhJq6WmDQv00feQM9V8E';
export const HELLOSIGN_CLIENT_ID = '530b26fda96d75b4abef002d9876fb7c';

export const ROLE_PRIMARY_APPLICANT = 'primary_applicant';
export const ROLE_CO_APPLICANT = 'co_applicant';
export const ROLE_GUARANTOR = 'guarantor';
export const ROLE_OCCUPANT = 'occupant';

export const SMS_OPT_IN_MARKETING_ENABLED = 5;

// event constants used for navigation
export const EVENT_APPLICANT_REGISTERED = 5;
export const EVENT_LEASE_TERMS_COMPLETED = 6;

export const EVENT_RENTAL_OPTIONS_SELECTED = 10;
export const EVENT_RENTAL_OPTIONS_NOT_SELECTED = 15;
export const EVENT_RENTAL_OPTIONS_COAPPLICANT_INVITED = 20;
export const EVENT_RENTAL_OPTIONS_GUARANTOR_INVITED = 30;
export const EVENT_RENTAL_OPTIONS_PET_ADDED = 40;

export const EVENT_INCOME_REPORTS_GENERATED = 45;
export const EVENT_APPLICATION_FEE_PAID = 50;

export const EVENT_SCREENING_COMPLETED = 60;
export const EVENT_INCOME_REPORTED_NONE = 185;
export const EVENT_FINANCIAL_STREAM_ADDITIONAL_DOCUMENTS_REQUESTED_EMAIL_SENT = 390;
export const EVENT_APPLICANT_UPDATED_EMPLOYER_INFO = 400;

// Applicant Milestones
export const MILESTONE_RENTAL_PROFILE_COMPLETED = 10020;
export const MILESTONE_INCOME_COMPLETED = 10030;
export const MILESTONE_APPLICATION_FEE_COMPLETED = 10040;
export const MILESTONE_HOLDING_DEPOSIT_SIGNED = 10050;
export const MILESTONE_APPLICANT_NEEDS_TO_REAGREE_TO_HD = 10065;
export const MILESTONE_APPLICANT_SUBMITTED = 10070;
export const MILESTONE_FINANCIAL_STREAM_INCOMPLETE = 10080;
export const MILESTONE_FINANCIAL_STREAM_ADDITIONAL_DOCUMENTS_REQUESTED = 10090;
export const MILESTONE_FINANCIAL_STREAM_MISSING_DOCUMENTS_REQUESTED = 10110;
export const MILESTONE_APPLICANT_SIGNED_LEASE = 10100;
// Application Milestones
export const MILESTONE_REQUEST_GUARANTOR = 20020;
export const MILESTONE_LEASE_SENT = 20030;
export const MILESTONE_LEASE_VOIDED = 20040;

export const APPLICATION_STATUS_APPROVED = 10;
export const APPLICATION_STATUS_DENIED = 15;
export const APPLICATION_STATUS_CANCELED = 20;
export const APPLICATION_STATUS_COMPLETED = 25;
export const APPLICATION_STATUS_CONDITIONALLY_APPROVED = 30;

export const APPLICANT_EVENTS = {
    // Regular events
    EVENT_APPLICANT_REGISTERED,
    EVENT_LEASE_TERMS_COMPLETED,
    EVENT_RENTAL_OPTIONS_SELECTED,
    EVENT_RENTAL_OPTIONS_NOT_SELECTED,
    EVENT_RENTAL_OPTIONS_COAPPLICANT_INVITED,
    EVENT_RENTAL_OPTIONS_GUARANTOR_INVITED,
    EVENT_RENTAL_OPTIONS_PET_ADDED,
    EVENT_INCOME_REPORTS_GENERATED,
    EVENT_APPLICATION_FEE_PAID,
    EVENT_SCREENING_COMPLETED,
    EVENT_APPLICANT_UPDATED_EMPLOYER_INFO,
    EVENT_INCOME_REPORTED_NONE,
    EVENT_FINANCIAL_STREAM_ADDITIONAL_DOCUMENTS_REQUESTED_EMAIL_SENT,

    // Milestones
    MILESTONE_APPLICANT_SIGNED_LEASE,
    MILESTONE_LEASE_VOIDED,
    MILESTONE_INCOME_COMPLETED,
    MILESTONE_HOLDING_DEPOSIT_SIGNED,
    MILESTONE_REQUEST_GUARANTOR,
    MILESTONE_APPLICANT_NEEDS_TO_REAGREE_TO_HD,
};

export const APPLICATION_STATUSES = {
    APPLICATION_STATUS_APPROVED,
    APPLICATION_STATUS_CONDITIONALLY_APPROVED,
    APPLICATION_STATUS_COMPLETED,
    APPLICATION_STATUS_DENIED,
    APPLICATION_STATUS_CANCELED,
};

// payments constants
export const LINE_ITEM_TYPE_APPLICATION_FEE = 10;
export const LINE_ITEM_TYPE_HOLDING_DEPOSIT = 20;

// applicant status constants
export const CO_APPLICANT_STATUS_NOT_STARTED = 'Not Started';
export const CO_APPLICANT_STATUS_IN_PROGRESS = 'In Progress';
export const CO_APPLICANT_STATUS_COMPLETED = 'Completed';

export const APPLICANT_STATUS_COLOR_MAP = {
    [CO_APPLICANT_STATUS_NOT_STARTED]: '#DB5963',
    [CO_APPLICANT_STATUS_IN_PROGRESS]: '#FAC700',
    [CO_APPLICANT_STATUS_COMPLETED]: '#67C18B',
};

// pets constants

export const RENTAL_OPTIONS_PETS_DOGS = 'pets-dogs';
export const RENTAL_OPTIONS_PETS_CATS = 'pets-cats';
export const RENTAL_OPTIONS_PETS_OTHER = 'pets-other';
export const PET_DOG_LABEL = 'Dog';
export const PET_CAT_LABEL = 'Cat';
export const PET_OTHER_LABEL = 'Other';

export const PET_RENTAL_OPTION_TYPE_TO_LABEL_MAP = {
    [RENTAL_OPTIONS_PETS_DOGS]: PET_DOG_LABEL,
    [RENTAL_OPTIONS_PETS_CATS]: PET_CAT_LABEL,
    [RENTAL_OPTIONS_PETS_OTHER]: PET_OTHER_LABEL,
};

export const RENTER_PROFILE_TYPE_CO_APPLICANTS = 'co_applicants';
export const RENTER_PROFILE_TYPE_OCCUPANT = 'occupant';
export const RENTER_PROFILE_TYPE_DEPENDENT = 'dependent';
export const RENTER_PROFILE_TYPE_GUARANTOR = 'guarantor';

export const RENTER_PROFILE_TYPE_PETS = 'pets';
export const RENTER_PROFILE_TYPE_PARKING = 'parking';
export const RENTER_PROFILE_TYPE_WINE_COOLER = 'wine-cooler';
export const RENTER_PROFILE_TYPE_STORAGE = 'storage';

// Accepted Terms Of Service Constants
export const TOS_TYPE_PAYMENTS = 'TOS_TYPE_PAYMENTS';
export const TOS_TYPE_NESTIO = 'TOS_TYPE_NESTIO';
export const TOS_TYPE_FINICITY = 'TOS_TYPE_FINICITY';

export const INCOME_TYPE_FINICITY_AUTOMATED = 135;

export const ALL_INCOME_OR_ASSET_TYPES = {
    // INCOME TYPES
    105: {
        label: 'Employment',
        value: 105,
    },
    110: {
        label: 'Self Employment',
        value: 110,
    },
    115: {
        label: 'Student',
        value: 115,
    },
    120: {
        label: 'Government',
        value: 120,
    },
    125: {
        label: 'Retirement',
        value: 125,
    },
    130: {
        label: 'Child/Spousal Support',
        value: 130,
    },
    140: {
        label: 'Other',
        value: 140,
    },
    // ASSET TYPES
    500: {
        label: 'Savings',
        value: 500,
    },
    505: {
        label: 'Retirement',
        value: 505,
    },
    510: {
        label: 'Investment',
        value: 510,
    },
    515: {
        label: 'Other',
        value: 515,
    },
};
// used in lists
export const INCOME_TYPES = [
    ALL_INCOME_OR_ASSET_TYPES[105],
    ALL_INCOME_OR_ASSET_TYPES[110],
    ALL_INCOME_OR_ASSET_TYPES[115],
    ALL_INCOME_OR_ASSET_TYPES[120],
    ALL_INCOME_OR_ASSET_TYPES[125],
    ALL_INCOME_OR_ASSET_TYPES[130],
    ALL_INCOME_OR_ASSET_TYPES[140],
];

export const ASSET_TYPES = [
    ALL_INCOME_OR_ASSET_TYPES[500],
    ALL_INCOME_OR_ASSET_TYPES[505],
    ALL_INCOME_OR_ASSET_TYPES[510],
    ALL_INCOME_OR_ASSET_TYPES[515],
];

export const FINANCIAL_STREAM_INCOME = 5;
export const FINANCIAL_STREAM_ASSET = 10;

export const FINANCIAL_STREAM_STATUS_PENDING = 10;
export const FINANCIAL_STREAM_STATUS_APPROVED = 20;
export const FINANCIAL_STREAM_STATUS_DENIED = 30;
export const FINANCIAL_STREAM_STATUS_INCOMPLETE = 40;

export const INCOME_TYPE_OTHER = 140;
export const ASSET_TYPE_OTHER = 515;

export const DOCUMENT_TYPE_LEASE = 5;
export const DOCUMENT_TYPE_HOLDING_DEPOSIT = 10;

export const DOCUMENT_TYPE_OPTIONS = { DOCUMENT_TYPE_LEASE, DOCUMENT_TYPE_HOLDING_DEPOSIT };

export const PAYMENT_TIME_MONTHLY = 10;
