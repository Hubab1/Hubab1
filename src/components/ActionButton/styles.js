import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
