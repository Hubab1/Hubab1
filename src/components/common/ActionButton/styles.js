import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
