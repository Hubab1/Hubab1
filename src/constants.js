export const community = {
    name: '555 Waverly'
};

export const Routes = {
    WELCOME: 'welcome',
    PROFILE: 'profile',
    LOGIN: 'login',
    SIGNUP: 'signup',
    TOS: 'terms',
    PROFILE_OPTIONS: 'profile/options',
    PETS: 'profile/pets'
}

export const ALL_ROUTES = [
    Routes.LOGIN, Routes.TOS, Routes.PROFILE_OPTIONS, Routes.PETS
]
