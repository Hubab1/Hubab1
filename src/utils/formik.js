// Ensure all fields are initialized in initialValues
export function allValuesSet (values, options={}) {
    let exclude = new Set();
    if (options.exclude) exclude = new Set(options.exclude);
    for (let key in values) {
        if (exclude.has(key)) continue;
        if (values[key] == null || values[key] === '') return false;
        if ((key === 'phone_number') && (values[key] === '(___) ___-____')) return false;
    }
    return true;
}
