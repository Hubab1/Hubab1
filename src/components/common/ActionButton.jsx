import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}