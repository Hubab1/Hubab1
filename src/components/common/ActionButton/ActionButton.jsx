import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}