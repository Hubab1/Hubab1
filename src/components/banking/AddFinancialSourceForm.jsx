import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import * as Yup from 'yup';
import CurrencyTextField from '@unicef/material-ui-currency-textfield';
import capitalize from 'lodash/capitalize';

import { allValuesSet } from 'utils/formik';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { Spacer } from 'assets/styles';
import { ASSET_TYPES, INCOME_TYPES, FINANCIAL_STREAM_ASSET } from 'app/constants';
import { Formik } from 'formik';

export default function AddFinancialSourceForm (props) {
    const financialTypeLabel = props.financialType === FINANCIAL_STREAM_ASSET ? 'asset' : 'income';
    const selectChoices = props.financialType === FINANCIAL_STREAM_ASSET ?
        ASSET_TYPES : INCOME_TYPES;

    function getInitialValues () {
        return Object.assign({
            income_or_asset_type: '',
            estimated_amount: ''
        }, props.initialValues);
    }
    return (
        <Formik
            validationSchema={
                Yup.object({
                    income_or_asset_type: Yup.number().required('Required'),
                    estimated_amount: Yup.string().required('Required')
                })
            }
            onSubmit={props.onSubmit}
            initialValues={getInitialValues()}
        >
            {
            ({
                values,
                handleChange,
                handleSubmit,
                errors,
                submitCount,
                isSubmitting
            }) => (
            <form onSubmit={handleSubmit}>
                <div className="align-left">
                    <FormControl fullWidth>
                        <InputLabel htmlFor="income-or-asset-type">{financialTypeLabel} type</InputLabel>
                        <Select
                            error={!!errors.income_or_asset_type}
                            value={values.income_or_asset_type}
                            fullWidth
                            onChange={handleChange}
                            inputProps={{
                                name: 'income_or_asset_type',
                                id: 'income-or-asset-type',
                            }}
                        >
                            {selectChoices.map(incomeType => (
                                <MenuItem key={incomeType.value} value={incomeType.value}>{incomeType.label}</MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                    <Spacer height={24}/>
                    {
                        values.income_or_asset_type &&
                        <CurrencyTextField
                            error={submitCount > 0 && !!errors.estimated_amount}
                            helperText={submitCount > 0 && errors.estimated_amount}
                            fullWidth
                            textAlign="left"
                            label={`Estimated ${financialTypeLabel} balance`}
                            minimumValue="0"
                            name="estimated_amount"
                            currencySymbol="$"
                            outputFormat="string"
                            onChange={handleChange}
                        />
                    }
                </div>
                <ActionButton disabled={!allValuesSet(values) || isSubmitting} marginTop={40} marginBottom={20}>
                    Add {capitalize(financialTypeLabel)} Source
                </ActionButton>
            </form>
            )}
        </Formik>
    )
}
