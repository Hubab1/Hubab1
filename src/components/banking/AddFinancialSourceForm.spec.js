import React from 'react';
import { shallow } from 'enzyme';
import { Formik } from 'formik';

import { INCOME_TYPE_OTHER } from 'app/constants';
import ActionButton from 'components/common/ActionButton/ActionButton';
import API from 'app/api';
import { AddFinancialSourceForm } from './AddFinancialSourceForm';


let defaultProps;
beforeEach(() => {
    defaultProps = {
        history: {
            push: jest.fn()
        },
        initialValues: {
            income_or_asset_type: 10,
            estimated_amount: '1111'
        }
    }
});

it('Disables submit if all values not set', async () => {
    const wrapper = shallow(<AddFinancialSourceForm {...defaultProps} initialValues={undefined}/>);
    expect(wrapper.find(Formik).dive().find(ActionButton).prop('disabled')).toBe(true);
});

it('Matches snapshot showing other field', async () => {
    const form = shallow(<AddFinancialSourceForm {...defaultProps} initialValues={{income_or_asset_type: INCOME_TYPE_OTHER}}/>).find(Formik).dive().find('form');
    expect(form.getElement()).toMatchSnapshot();
});
