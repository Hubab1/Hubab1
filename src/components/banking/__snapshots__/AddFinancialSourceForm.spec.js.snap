// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot showing other field 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="align-left"
  >
    <ForwardRef(WithStyles)
      fullWidth={true}
    >
      <ForwardRef(WithStyles)
        htmlFor="income-or-asset-type"
      >
        Income
         type
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        error={false}
        fullWidth={true}
        inputProps={
          Object {
            "id": "income-or-asset-type",
            "name": "income_or_asset_type",
          }
        }
        onChange={[Function]}
        value={140}
      >
        <ForwardRef(WithStyles)
          value={105}
        >
          Employment
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={110}
        >
          Self Employment
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={115}
        >
          Student
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={120}
        >
          Government
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={125}
        >
          Retirement
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={130}
        >
          Child/Spousal Support
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          value={140}
        >
          Other
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <React.Fragment>
      <ForwardRef(render)
        height={24}
      />
      <FormTextInput
        handleChange={[Function]}
        inputProps={
          Object {
            "maxLength": 255,
          }
        }
        label="Description"
        name="other"
        submitted={false}
        type="text"
      />
    </React.Fragment>
    <ForwardRef(render)
      height={24}
    />
    <React.Fragment>
      <ForwardRef(WithStyles)
        currencySymbol="$"
        error={false}
        fullWidth={true}
        helperText={false}
        label="Estimated annual income"
        minimumValue="0"
        name="estimated_amount"
        onChange={[Function]}
        outputFormat="string"
        textAlign="left"
        value=""
      />
      <UNDEFINED
        incomeOrAssetType={140}
      />
    </React.Fragment>
  </div>
  <ActionButton
    color="primary"
    disabled={true}
    marginBottom={20}
    marginTop={40}
    successGreen={false}
    variant="contained"
  >
    Add Income Source
  </ActionButton>
</form>
`;
